pkg/massdns/process.go

```diff
@@ -212,8 +213,26 @@ func (c *Client) writeOutput(store *store.Store) error {
 			}
 			uniqueMap[hostname] = struct{}{}
 
-			buffer.WriteString(hostname)
-			buffer.WriteString("\n")
+			if c.config.Json {
```

Comment:
I think this could be simplified and marshaled only one time:

```
if c.config.Json {
	hostnameJson, err := json.Marshal(map[string]interface{}{"hostname": hostname})
	if err != nil {
		return fmt.Errorf("could not marshal output as json: %v", err)
	}
	buffer.WriteString(string(hostnameJson))
	buffer.WriteString("\n")
} else {
	buffer.WriteString(hostname)
	buffer.WriteString("\n")
}
```

Reply:
Thanks!